### ISSUE TITLE
About SIDA-13B Model

### ISSUE THREAD

**Issue by Jiacheng8** [hzlsaber/SIDA#22]
Dear Author,

Thank you very much for your excellent work! I encountered an issue while fine-tuning the **SIDA-13B** model on the **SIDA-training** dataset , the observed losses seem quite high for the first epoch. Is this expected, since SIDA-13B should already be fine-tuned on the SIDA training set?

```log
Epoch: [0][   1/1000]   Time 13.309 (13.309)    Loss 1.1543 (2.9163)    ClsLoss 1.1543 (1.1122) MaskBCELoss 1.9652 (1.8040)     MaskDICELoss 1.9652 (1.8040)   MaskLoss 3.9304 (3.6081)
```

Additionally, I noticed that the **BCE loss** and **DICE loss** values are identical. Upon checking the code, I found they were initialized as:

```python
mask_bce_loss = mask_dice_loss = mask_loss = torch.tensor(0.0, device=cls_loss.device)
```

This makes me wonder whether the mask losses are being computed correctly. Could you please confirm if this behavior is intentional or a potential bug?

**Comment by hzlsaber** [comment]
Thank you for your kind words and for carefully checking the code!

1. The higher initial loss is expected — SIDA-13B is initialized from LISA-13B-llama2-v1 and has not been fine-tuned on the SIDA training set. Only the “SIDA-*-description” models are further fine-tuned on the 3K description subset, so the first few epochs naturally show larger losses before convergence.

2. The line: mask_bce_loss = mask_dice_loss = mask_loss = torch.tensor(0.0, device=cls_loss.device)

is only an initialization placeholder. The BCE and DICE losses are computed separately later in the code. If you see identical numbers in logs, it’s likely due to how the logging statement is written rather than an actual bug. You can separate the initialization lines to avoid confusion.

**Comment by Jiacheng8** [comment]
Thank you very much for the clarification!
That makes sense regarding the initialization and logging part.

I just have a small follow-up question:
In the paper (Section 5, Experiments), it’s stated that “We fine-tuned both LISA-7B-v1 and LISA-13B-v1 on the SID-Set using LoRA…”, which suggests that SIDA-13B had already been fine-tuned on the SID-Set.

Could you please confirm whether the SIDA-13B model used in the paper’s experiments was indeed fine-tuned on the SID-Set, while the released SIDA-13B checkpoint is the pre-fine-tuned version initialized from LISA-13B?

Just want to make sure I’m aligning with the setup used in the paper.
Thanks again for your time and for sharing this great work!

**Comment by hzlsaber** [comment]
In the paper, both SIDA-7B and SIDA-13B were fine-tuned from LISA-7B/13B-v1 on the SID-Set using LoRA, and the released checkpoints correspond to those fine-tuned versions.

If you fine-tune the released SIDA-13B again on the same dataset, it’s normal to observe relatively high losses at the beginning. The optimizer and LoRA layers are re-initialized, and the learning rate warm-up causes temporary fluctuations. The loss should quickly stabilize after a few hundred iterations.

Best and hope your work goes well.
